resolver/resolver.cc

```diff
@@ -1683,6 +1705,29 @@ class ResolveConstantsWalk {
         return depth;
     }
 
+    static core::ClassOrModuleRef resolvePayloadSuperclassClassName(const core::GlobalState &gs,
```

Comment:
Would it make sense to add a comment explaining this is not a technique we expect people to use?
```suggestion
    // This function resolves a fully qualified class name into a ClassOrModuleRef.
    // Usually in Sorbet, we steer away from string matching in favor of using `SymbolRef`s due to performance benefits.
    // However, this function stands as an exception to assist with the `--suppress-payload-superclass-redefinition-for`
    // command. Please, do not repeat this practice elsewhere.
    static core::ClassOrModuleRef resolvePayloadSuperclassClassName(const core::GlobalState &gs,
```

Reply:
If you have strings, this is the preferred way of doing things.

> Usually in Sorbet, we steer away from string matching in favor of using `SymbolRef`s due to performance benefits.

The quoted piece of code _does not_ use string matching—it explicitly converts from a String to a SymbolRef, once per input string. It's as close as you could get to parsing the class name as a ruby program and resolving the corresponding constant lit.

If someone else needed to do this elsewhere in Sorbet, this would absolutely be the code worth copying referencing, and the name of the function should make it clear enough that this function is not meant to be used for anything except for this PayloadSuperclass error thing (if someone tried to call this exact function elsewhere, it would be easy to catch that in code review).